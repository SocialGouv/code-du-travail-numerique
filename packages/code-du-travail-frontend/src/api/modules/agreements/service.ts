import { SearchResponse, ElasticSearchItem, Agreement } from "cdtn-types";
import { elasticsearchClient, elasticIndex, NotFoundError } from "../../utils";
import {
  getAllAgreementsWithContributions,
  getAgreementsBySlugs,
  getAgreementBySlugBody,
  getAgreementsByIds,
} from "./queries";

export const getAllAgreements = async (): Promise<Agreement[]> => {
  const body = getAllAgreementsWithContributions();

  const response = await elasticsearchClient.search<SearchResponse<Agreement>>({
    body,
    index: elasticIndex,
  });

  return response.body.hits.hits
    .map(({ _source }) => _source)
    .sort(orderByAlphaAndMetalurgieLast);
};

export const getBySlugsAgreements = async (
  slugs: string[]
): Promise<ElasticSearchItem[]> => {
  const body = getAgreementsBySlugs(slugs);
  const response = await elasticsearchClient.search({
    body,
    index: elasticIndex,
  });
  return response.body.hits.total.value > 0
    ? response.body.hits.hits.map(({ _source }) => _source)
    : [];
};

export const getByIdsAgreements = async (
  ids: string[]
): Promise<ElasticSearchItem[]> => {
  const body = getAgreementsByIds(ids);
  const response = await elasticsearchClient.search({
    body,
    index: elasticIndex,
  });
  return response.body.hits.total.value > 0
    ? response.body.hits.hits.map(({ _source }) => _source)
    : [];
};

export const getBySlugAgreements = async (slug: string) => {
  const body = await getAgreementBySlugBody(slug);

  const response = await elasticsearchClient.search({
    body,
    index: elasticIndex,
  });
  if (response.body.hits.total.value === 0) {
    throw new NotFoundError({
      message: `Agreement not found, no agreement match ${slug}y`,
      name: "AGREEMENT_NOT_FOUND",
      cause: null,
    });
  }

  return { ...response.body.hits.hits[0]._source };
};

const orderByAlphaAndMetalurgieLast = (a, b) => {
  if (a.url && !b.url) {
    return -1;
  }
  if (!a.url && b.url) {
    return 1;
  }
  if (a.title < b.title) {
    return -1;
  }
  if (a.title > b.title) {
    return 1;
  }

  return 0;
};
