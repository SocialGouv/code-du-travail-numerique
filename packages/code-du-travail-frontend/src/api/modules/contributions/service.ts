import { ElasticSearchItem } from "@socialgouv/cdtn-utils";
import {
  elasticsearchClient,
  elasticDocumentsIndex,
  NotFoundError,
} from "../../utils";
import {
  getAllGenericsContributions,
  getContributionsBySlugs,
  getAllContributionBySlug,
  getContributionsByIds,
} from "./queries";

export const getGenericContributionsGroupByThemes = async () => {
  const body = getAllGenericsContributions();

  const response = await elasticsearchClient.search({
    body,
    index: elasticDocumentsIndex,
  });
  return response.body.hits.hits
    .map(({ _source }) => _source)
    .map((contrib) => {
      contrib.theme = contrib.breadcrumbs[0].label;
      return contrib;
    })
    .reduce(groupByThemes, {});
};

export const getGenericsContributions = async () => {
  const body = getAllGenericsContributions();

  const response = await elasticsearchClient.search({
    body,
    index: elasticDocumentsIndex,
  });
  return response.body.hits.hits.map(({ _source }) => _source);
};

export const getBySlugsContributions = async (
  slugs: string[]
): Promise<ElasticSearchItem[]> => {
  const body = getContributionsBySlugs(slugs);
  const response = await elasticsearchClient.search({
    body,
    index: elasticDocumentsIndex,
  });
  return response.body.hits.total.value > 0
    ? response.body.hits.hits.map(({ _source }) => _source)
    : [];
};

export const getByIdsContributions = async (
  ids: string[]
): Promise<ElasticSearchItem[]> => {
  const body = getContributionsByIds(ids);
  const response = await elasticsearchClient.search({
    body,
    index: elasticDocumentsIndex,
  });
  return response.body.hits.total.value > 0
    ? response.body.hits.hits.map(({ _source }) => _source)
    : [];
};

const groupByThemes = (acc, item) => {
  if (item.theme in acc) acc[item.theme].push(item);
  else acc[item.theme] = [item];
  return acc;
};
