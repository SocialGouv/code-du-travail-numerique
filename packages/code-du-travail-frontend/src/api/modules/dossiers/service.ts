import { SOURCES } from "@socialgouv/cdtn-utils";
import {
  elasticsearchClient,
  elasticDocumentsIndex,
  NotFoundError,
} from "../../utils";
import { getSearchBySourceSlugBody } from "../items/queries";

export const getDossiers = async (slug: string) => {
  const body = getSearchBySourceSlugBody({
    slug,
    source: SOURCES.THEMATIC_FILES,
  });
  const response = await elasticsearchClient.search({
    body,
    index: elasticDocumentsIndex,
  });

  if (response.body.hits.total.value === 0) {
    throw new NotFoundError({
      message: `There is no thematic files that match ${slug}`,
      name: "THEMATIC_FILES_NOT_FOUND",
      cause: null,
    });
  }

  const thematicFile = response.body.hits.hits[0]._source;

  return {
    ...thematicFile,
  };
};
