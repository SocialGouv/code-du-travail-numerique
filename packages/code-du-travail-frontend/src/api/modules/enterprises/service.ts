import { SearchResponse, Agreement } from "@socialgouv/cdtn-utils";
import { elasticDocumentsIndex, elasticsearchClient } from "../../utils";
import { getAgreements } from "./queries";
import { EnterpriseApiResponse, ApiEnterpriseData, Convention } from "./types";
import { IDCC_TO_REPLACE } from "../../config";

const toAgreement = (convention: Convention): Agreement => ({
  id: convention.id ?? convention.idcc,
  num: convention.idcc,
  shortTitle: convention.shortTitle ?? "Convention collective non reconnue",
  title: convention.title,
  ...(convention.url ? { url: convention.url } : {}),
});

export const populateAgreements = async (
  enterpriseApiResponse: EnterpriseApiResponse
): Promise<ApiEnterpriseData> => {
  const idccList: number[] =
    enterpriseApiResponse.entreprises?.flatMap((enterprise) =>
      enterprise.conventions.flatMap((convention) => convention.idcc)
    ) ?? [];
  const idccKeys = Object.keys(IDCC_TO_REPLACE);
  const idcc = idccList.find((idcc) => idccKeys.includes(idcc.toString()));
  if (idcc) {
    const idccToAdd = IDCC_TO_REPLACE[idcc] as number[];
    idccList.push(...idccToAdd);
  }

  if (idccList.length > 0) {
    const body = getAgreements(idccList);
    const response = await elasticsearchClient.search<
      SearchResponse<Agreement>
    >({ body, index: elasticDocumentsIndex });

    if (response.body.hits.total.value > 0) {
      const agreements = response.body.hits.hits.reduce(
        (acc: Record<number, Agreement>, curr) => {
          acc[curr._source.num] = curr._source;
          return acc;
        },
        {}
      );

      const result = {
        ...enterpriseApiResponse,
        entreprises: enterpriseApiResponse.entreprises?.map((enterprise) => ({
          ...enterprise,
          conventions: enterprise.conventions.map(
            (convention): Agreement =>
              agreements[convention.idcc] ?? toAgreement(convention)
          ),
        })),
      };

      if (idcc) {
        const idccToAdd = IDCC_TO_REPLACE[idcc] as number[];
        let conventionsToAdd: Agreement[] = idccToAdd
          .map((idcc) => agreements[idcc])
          .filter((convention) => convention !== undefined);
        result.entreprises = result.entreprises?.map((enterprise) => {
          const hasAlreadyIdcc = enterprise.conventions.find((convention) =>
            idccToAdd.includes(convention.num)
          );
          if (hasAlreadyIdcc) {
            conventionsToAdd = [];
          }
          return {
            ...enterprise,
            conventions: [...enterprise.conventions, ...conventionsToAdd]
              .filter((convention) => convention.num !== idcc)
              .filter(
                (convention, index, self) => self.indexOf(convention) === index
              ),
          };
        });
      }

      return result;
    }
  }
  return {
    ...enterpriseApiResponse,
    entreprises: enterpriseApiResponse.entreprises?.map((enterprise) => ({
      ...enterprise,
      conventions: enterprise.conventions.map(toAgreement),
    })),
  };
};
