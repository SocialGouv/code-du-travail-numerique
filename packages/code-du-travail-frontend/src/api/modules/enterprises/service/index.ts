export * from "./fetchEnterprises";
export * from "./populateAgreements";
