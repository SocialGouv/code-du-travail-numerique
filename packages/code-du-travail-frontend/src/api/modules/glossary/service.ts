import memoizee from "memoizee";
import {
  elasticDocumentsIndex,
  elasticsearchClient,
  NotFoundError,
} from "../../utils";
import { getGlossaryBody } from "./queries";

async function getGlossaryData() {
  const body = getGlossaryBody();

  const response = await elasticsearchClient.search({
    body,
    index: elasticDocumentsIndex,
  });

  if (response.body.hits.total.value === 0) {
    throw new NotFoundError({
      message: `There is no glossary data`,
      cause: null,
      name: "GLOSSARY_NOT_FOUND",
    });
  }
  const glossaryData = response.body.hits.hits[0]._source.data;

  return glossaryData;
}

export const getGlossary = memoizee(getGlossaryData, {
  maxAge: 1000 * 5 * 60,
  preFetch: true,
  promise: true,
});
