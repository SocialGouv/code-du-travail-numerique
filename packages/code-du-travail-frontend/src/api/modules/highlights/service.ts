import {
  elasticsearchClient,
  elasticDocumentsIndex,
  NotFoundError,
} from "../../utils";
import { getHighlightsBySlug } from "./queries";

export const getBySlugHighlights = async (slug: string) => {
  const body = getHighlightsBySlug(slug);

  const response = await elasticsearchClient.search<any>({
    body,
    index: elasticDocumentsIndex,
  });

  if (response.hits.hits.length === 0) {
    throw new NotFoundError({
      message: `There is no highlight that match ${slug}`,
      name: "HIGHLIGHT_NOT_FOUND",
      cause: null,
    });
  }

  return response.hits.hits[0]._source.refs;
};
