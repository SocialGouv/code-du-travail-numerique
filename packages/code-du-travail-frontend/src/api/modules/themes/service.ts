import {
  elasticsearchClient,
  elasticDocumentsIndex,
  NotFoundError,
} from "../../utils";
import {
  getAllThemesQuery,
  getThemeBySlugQuery,
  getThemeBySlugsQuery,
} from "./queries";

export const getAllThemes = async () => {
  const body: any = getAllThemesQuery();
  const response = await elasticsearchClient.search<any>({
    body,
    index: elasticDocumentsIndex,
  });
  return {
    children: response.hits.hits.map((t) => t._source),
  };
};

export const getAllThemesAndSubThemes = async () => {
  const body: any = getAllThemesQuery();
  const response = await elasticsearchClient.search<any>({
    body,
    index: elasticDocumentsIndex,
  });
  const themes = response.hits.hits.map((t) => t._source);
  // for each theme of themes, we need to get slug of children
  const childrenSlugs = themes.flatMap((theme) =>
    theme.children.map((child) => child.slug)
  );
  const data = await getBySlugsThemes(childrenSlugs).catch(() => {
    return [];
  });
  const themesWithChildren = themes.map((theme) => {
    const children = theme.children.map((child) => {
      const childWithContent = data.find((d: any) => d.slug === child.slug);
      return {
        ...child,
        ...childWithContent,
      };
    });
    return {
      ...theme,
      children,
    };
  });
  return themesWithChildren;
};

export const getBySlugThemes = async (slug: string) => {
  const body: any = getThemeBySlugQuery(slug);

  const response = await elasticsearchClient.search<any>({
    body,
    index: elasticDocumentsIndex,
  });

  if (response.hits.hits.length === 0) {
    throw new NotFoundError({
      message: `There is no theme that match ${slug}`,
      name: "THEME_NOT_FOUND",
      cause: null,
    });
  }

  const theme = response.hits.hits[0];

  return {
    ...theme._source,
  };
};

export const getBySlugsThemes = async (slugs: string[]) => {
  const body: any = getThemeBySlugsQuery(slugs);

  const response = await elasticsearchClient.search<any>({
    body,
    index: elasticDocumentsIndex,
  });

  if (response.hits.hits.length === 0) {
    throw new NotFoundError({
      message: `There is no theme that match ${slugs.join(",")}`,
      name: "THEME_NOT_FOUND",
      cause: null,
    });
  }

  const themes = response.hits.hits.map(({ _source }) => _source);

  return themes;
};
