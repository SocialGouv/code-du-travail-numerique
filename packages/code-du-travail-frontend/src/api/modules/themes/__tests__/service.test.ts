import { getAllThemes, getBySlugThemes } from "../service";

describe("Themes", () => {
  it("getAllThemes", async () => {
    const result = await getAllThemes();
    expect(result).toMatchSnapshot();
  });

  it("getBySlugThemes", async () => {
    const result = await getBySlugThemes("embauche-et-contrat-de-travail");
    expect(result).toMatchSnapshot();
  });
});
