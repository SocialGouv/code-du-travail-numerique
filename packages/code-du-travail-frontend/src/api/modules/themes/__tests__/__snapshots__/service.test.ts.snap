// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Themes getAllThemes 1`] = `
{
  "children": [
    {
      "children": [
        {
          "label": "Embauche",
          "slug": "embauche",
        },
        {
          "label": "Contrat de travail",
          "slug": "contrat-de-travail",
        },
      ],
      "icon": "Contract",
      "position": 1,
      "slug": "embauche-et-contrat-de-travail",
      "title": "Embauche et contrat de travail",
    },
    {
      "children": [
        {
          "label": "Documents à remettre au salarié",
          "slug": "documents-a-remettre-au-salarie",
        },
        {
          "label": "Démission",
          "slug": "demission",
        },
        {
          "label": "Fin d’un CDD - CTT",
          "slug": "fin-dun-cdd-ctt",
        },
        {
          "label": "Rupture conventionnelle",
          "slug": "rupture-conventionnelle",
        },
        {
          "label": "Licenciement : droits des salariés et procédures",
          "slug": "licenciement-droits-des-salaries-et-procedures",
        },
        {
          "label": "Retraite",
          "slug": "retraite",
        },
        {
          "label": "Autres départs",
          "slug": "autres-departs",
        },
      ],
      "icon": "Depart",
      "position": 8,
      "slug": "depart-de-lentreprise",
      "title": "Départ de l’entreprise",
    },
  ],
}
`;

exports[`Themes getAllThemesAndSubThemes 1`] = `
[
  {
    "children": [
      {
        "label": "Embauche",
        "slug": "embauche",
      },
      {
        "label": "Contrat de travail",
        "slug": "contrat-de-travail",
      },
    ],
    "icon": "Contract",
    "position": 1,
    "slug": "embauche-et-contrat-de-travail",
    "title": "Embauche et contrat de travail",
  },
  {
    "children": [
      {
        "label": "Documents à remettre au salarié",
        "slug": "documents-a-remettre-au-salarie",
      },
      {
        "breadcrumbs": [
          {
            "label": "Départ de l’entreprise",
            "position": 8,
            "slug": "/themes/depart-de-lentreprise",
          },
        ],
        "children": [],
        "label": "Démission",
        "position": 1,
        "refs": [
          {
            "cdtnId": "5d4a0ee541",
            "description": "La démission permet au salarié de rompre son CDI de sa propre initiative. Si un délai de préavis est prévu, il doit être respecté.",
            "slug": "la-demission",
            "source": "page_fiche_ministere_travail",
            "title": "La démission",
          },
          {
            "cdtnId": "8c42435f9a",
            "description": "La démission permet au salarié, à son initiative, de rompre son contrat de travail, sous conditions.",
            "slug": "demission-dun-salarie",
            "source": "fiches_service_public",
            "title": "Démission d'un salarié",
          },
          {
            "cdtnId": "a6773953e6",
            "description": "Le salarié peut revenir sur sa démission. La rétractation s'impose à l'employeur si la volonté réelle de démissionner du salarié est remise en cause.",
            "slug": "le-salarie-peut-il-revenir-sur-sa-demission",
            "source": "fiches_service_public",
            "title": "Le salarié peut-il revenir sur sa démission ?",
          },
          {
            "cdtnId": "db4f1fe3fb",
            "description": "La durée du préavis de démission est fixée :

Soit par une convention collective ou un accord collectif

Soit par les usages pratiqués dans la localité…",
            "slug": "quelle-est-la-duree-du-preavis-en-cas-de-demission",
            "source": "contributions",
            "title": "Quelle est la durée du préavis en cas de démission ?",
          },
          {
            "cdtnId": "1eea193273",
            "description": "Calculez la durée de préavis à respecter en cas de démission",
            "displayTool": true,
            "order": 1,
            "slug": "preavis-demission",
            "source": "outils",
            "title": "Préavis de démission",
          },
          {
            "cdtnId": "9fcdb4215e",
            "description": "Le préavis de démission doit être exécuté en totalité par l’employeur et le salarié, y compris si le salarié a retrouvé un emploi. Mais, le préavis n’est…",
            "slug": "le-preavis-de-demission-doit-il-etre-execute-en-totalite-y-compris-si-le-salarie-a-retrouve-un-emploi",
            "source": "contributions",
            "title": "Le préavis de démission doit-il être exécuté en totalité ? Y compris si le salarié a retrouvé un emploi ?",
          },
          {
            "cdtnId": "fdaf586c20",
            "description": "Le salarié qui démissionne pour suivre la personne avec qui il vit en couple doit effectuer un préavis, sauf dispense de l'employeur.",
            "slug": "est-on-dispense-de-preavis-en-cas-de-demission-liee-a-la-mutation-du-conjoint",
            "source": "fiches_service_public",
            "title": "Est-on dispensé de préavis en cas de démission liée à la mutation du conjoint ?",
          },
          {
            "cdtnId": "6ed149796a",
            "description": "Le salarié ne bénéficie pas d'heures de recherche d'emploi pendant son préavis suite à sa démission, sauf dispositions conventionnelles.",
            "slug": "beneficie-t-on-dheures-de-recherche-demploi-pendant-le-preavis-de-demission",
            "source": "fiches_service_public",
            "title": "Bénéficie-t-on d'heures de recherche d'emploi pendant le préavis de démission ?",
          },
          {
            "cdtnId": "184ebe1ef6",
            "description": "Oui. Toute salariée peut démissionner pendant un congé maternité, dans les conditions habituelles prévues en matière de démission.",
            "slug": "peut-on-demissionner-pendant-un-conge-maternite",
            "source": "fiches_service_public",
            "title": "Peut-on démissionner pendant un congé maternité ?",
          },
          {
            "cdtnId": "6b17d70cd9",
            "description": "Le salarié peut démissionner pendant un arrêt maladie, sauf s'il est en période d'essai.",
            "slug": "peut-on-demissionner-pendant-un-arret-maladie",
            "source": "fiches_service_public",
            "title": "Peut-on démissionner pendant un arrêt maladie ?",
          },
          {
            "cdtnId": "4e0dedaa2e",
            "description": "Le salarié d'un particulier employeur qui souhaite démissionner doit informer son employeur et respecter un préavis. Il peut bénéficier d'une indemnité compensatrice de préavis et éventuellement d'une indemnité compensatrice de congés payés. L'employeur doit lui remettre des documents de fin de contrat.",
            "slug": "demission-du-salarie-a-domicile-employe-par-un-particulier",
            "source": "fiches_service_public",
            "title": "Démission du salarié à domicile employé par un particulier",
          },
          {
            "cdtnId": "38474d168c",
            "description": "L'assistante maternelle (ou assistant maternel) agréée qui souhaite rompre son contrat de travail doit informer le particulier employeur par écrit et respecter un préavis. À la fin de celui-ci, l'employeur remet les documents de fin de contrat.",
            "slug": "demission-dune-assistante-maternelle",
            "source": "fiches_service_public",
            "title": "Démission d'une assistante maternelle",
          },
          {
            "cdtnId": "503c412884",
            "description": "Le salarié peut démissionner pendant son congé parental, dans le respect de la procédure habituelle prévue en matière de démission.",
            "slug": "peut-on-demissionner-pendant-un-conge-parental",
            "source": "fiches_service_public",
            "title": "Peut-on démissionner pendant un congé parental ?",
          },
          {
            "cdtnId": "2e47dcbf91",
            "description": "Seules les personnes involontairement privées d'emploi peuvent bénéficier des allocations chômage. La démission n'ouvre pas de droit au chômage, sauf dérogation",
            "slug": "le-droit-aux-allocations-chomage-du-salarie-demissionnaire",
            "source": "page_fiche_ministere_travail",
            "title": "Le droit aux allocations chômage du salarié démissionnaire",
          },
          {
            "cdtnId": "b1a0a48265",
            "description": "À partir du 1er novembre, tous les salariés ayant au moins 5 ans d'activité salariée dans une ou plusieurs entreprises auront droit à l'assurance (...)",
            "slug": "de-nouveaux-droits-pour-les-salaries-demissionnaires",
            "source": "page_fiche_ministere_travail",
            "title": "De nouveaux droits pour les salariés démissionnaires",
          },
          {
            "cdtnId": "4e7e702fb5",
            "description": "Le salarié qui démissionne n'a pas droit aux allocations chômage (ARE), sauf cas particuliers (démission légitime, reliquat de droits...).",
            "slug": "un-salarie-peut-il-percevoir-lallocation-chomage-en-cas-de-demission",
            "source": "fiches_service_public",
            "title": "Un salarié peut-il percevoir l'allocation chômage en cas de démission ?",
          },
          {
            "cdtnId": "3e70fb9868",
            "description": "Peut-on percevoir l'allocation chômage en cas de démission ?",
            "slug": "peut-on-percevoir-lallocation-chomage-en-cas-de-demission",
            "source": "fiches_service_public",
            "title": "Peut-on percevoir l'allocation chômage en cas de démission ?",
          },
        ],
        "slug": "demission",
        "title": "Démission",
      },
      {
        "label": "Fin d’un CDD - CTT",
        "slug": "fin-dun-cdd-ctt",
      },
      {
        "label": "Rupture conventionnelle",
        "slug": "rupture-conventionnelle",
      },
      {
        "label": "Licenciement : droits des salariés et procédures",
        "slug": "licenciement-droits-des-salaries-et-procedures",
      },
      {
        "label": "Retraite",
        "slug": "retraite",
      },
      {
        "label": "Autres départs",
        "slug": "autres-departs",
      },
    ],
    "icon": "Depart",
    "position": 8,
    "slug": "depart-de-lentreprise",
    "title": "Départ de l’entreprise",
  },
]
`;

exports[`Themes getBySlugThemes 1`] = `
{
  "breadcrumbs": [],
  "children": [
    {
      "label": "Embauche",
      "slug": "embauche",
    },
    {
      "label": "Contrat de travail",
      "slug": "contrat-de-travail",
    },
  ],
  "description": "Vous venez de trouver un travail, vous embauchez un salarié ou un apprenti ? Vous souhaitez faire une déclaration préalable à l’embauche, rédiger un contrat de travail (CDI, CDD, travail temporaire, temps plein, temps partiel) ? Retrouvez toutes nos réponses ici.  ",
  "position": 1,
  "refs": [
    {
      "cdtnId": "c87486b7f0",
      "description": "Tout employeur, sauf s'il s'agit d'un particulier employeur, doit obligatoirement détenir plusieurs registres : registre unique du personnel, registres relatifs aux conditions du travail, registre du comité social et économique (CSE) et document unique d'évaluation des risques professionnels (DUER ou DUERP).",
      "slug": "registres-du-personnel-obligatoires",
      "source": "fiches_service_public",
      "title": "Registres du personnel obligatoires",
    },
    {
      "cdtnId": "59d6f71101",
      "description": "Il assure la transparence des emplois dans chaque établissement avec l'identification des salariés, les dates d'embauche et de départ, les contrats de travail.",
      "slug": "le-registre-unique-du-personnel",
      "source": "page_fiche_ministere_travail",
      "title": "Le registre unique du personnel",
    },
    {
      "cdtnId": "f958c8b035",
      "description": "Dans le contexte actuel sans précédent, les salariés inoccupés qui le souhaitent, peuvent être transférés provisoirement dans une entreprise confrontée (...)",
      "slug": "mise-a-disposition-temporaire-de-salaries-volontaires-entre-deux-entreprises",
      "source": "page_fiche_ministere_travail",
      "title": "Mise à disposition temporaire de salariés volontaires entre deux entreprises",
    },
  ],
  "slug": "embauche-et-contrat-de-travail",
  "title": "Embauche et contrat de travail",
}
`;
